### ISSUE TITLE
Logstash not closing close_wait connections.

### ISSUE THREAD

**Issue by sc-chad** [logstash-plugins/logstash-input-beats#178]
- Version: 
logstash 5.1.2 
logstash-input-filebeat: 3.1.12
- Operating System: Amazon Linux
- Config File (if you have sensitive info, please remove it): 
- Sample Data:
- Steps to Reproduce:

Similar issue to: https://github.com/elastic/logstash/issues/6300
Previous discussed issue: https://github.com/logstash-plugins/logstash-output-s3/issues/120

Issue: I believe we have narrowed it down to the filebeat plugin not closing all the connections when using an Amazon ELB.  I have tried lowering the default timeout to 60 seconds which didn't seem to work.

Example:
```
java      15038      logstash 2152u     IPv4         1449165042         0t0        TCP prod-graylog-processor-775e2946:lxi-evntsvc->ip-10-0-5-216.ec2.internal:16700 (CLOSE_WAIT)
java      15038      logstash 2153u     IPv4         1449165044         0t0        TCP prod-graylog-processor-775e2946:lxi-evntsvc->ip-10-0-5-216.ec2.internal:16702 (CLOSE_WAIT)
java      15038      logstash 2154u     IPv4         1449165046         0t0        TCP prod-graylog-processor-775e2946:lxi-evntsvc->ip-10-0-1-185.ec2.internal:14513 (CLOSE_WAIT)
```

Showing command and count: 
```
/usr/sbin/lsof | grep "logstash" | grep "CLOSE_WAIT" | wc -l
2173
```

**Comment by feichaohao** [comment]
How to solve the problem?

**Comment by feichaohao** [comment]
filebeat  set : 
spool_size: 1024
 idle_timeout: 5s

logstash set : 
beat{         port => 5044
        type => "log"
        add_field=>{"short_message"=>"product"}
        add_field=>{"host"=>"host"}
    }

**Comment by ph** [comment]
fixed in #191

**Comment by patsevanton** [comment]
Hello!
in which version of Logstash is this bug fixed?
